src/lib/hash/blake2/blake2b.cpp

```diff
@@ -181,26 +213,50 @@ void BLAKE2b::final_result(uint8_t output[])
    state_init();
    }
 
+Key_Length_Specification BLAKE2b::key_spec() const
+   {
+   return Key_Length_Specification(0, 64);
+   }
+
 std::string BLAKE2b::name() const
    {
    return "BLAKE2b(" + std::to_string(m_output_bits) + ")";
    }
 
 HashFunction* BLAKE2b::clone() const
    {
-   return new BLAKE2b(m_output_bits);
+   return new BLAKE2b(m_padded_key_buffer.data(), m_key_size, m_output_bits);
```

Comment:
This clones with the key set which is inconsistent with the rest of the library (`clone` is a terrible name given that it just copies the object type and not the object state, but changing it for just one type is more confusing than anything else)